src/controllers/loginpage.js

```diff
@@ -60,8 +60,15 @@ define(["apphost", "appSettings", "dom", "connectionManager", "loading", "cardSt
         var html = "";
         for (var i = 0; i < users.length; i++) {
             var user = users[i];
-            html += '<button type="button" class="card squareCard scalableCard squareCard-scalable"><div class="cardBox cardBox-bottompadded">';
-            html += '<div class="cardScalable">';
+
+            var cardBoxCssClass = "cardBox cardBox-bottompadded";
+            if (layoutManager.tv) {
+                cardBoxCssClass += " cardBox-focustransform";
```

Comment:
In the future I'd like to move all focus related styling to a single class called `show-focus` to speed up development. That way we can just search for a single class and find every instance, rather than all the custom classes that currently exist.

Reply:
Do you want me to replace _cardBox-focustransform_ to _show-focus_ (or even mark the top element with it)?
Then `components/cardbuilder/card.css` with backward compatibility (not tested):
```
.cardBox-focustransform,
.cardBox.show-focus,           /* or */
.card.show-focus .cardBox { /* in case of marking the top element */
    will-change: transform;
    transition: transform 200ms ease-out;
}

.card:focus > .cardBox-focustransform,
.card:focus > .cardBox.show-focus,  /* or */
.card.show-focus:focus > .cardBox { /* in case of marking the top element */
    transform: scale(1.18);
}
```
Class _card-focuscontent_ adds some style by theme.

Reply:
I like the `.cardBox.show-focus` method best. Also, there's no need to maintain backward compatibility since we are changing all instances of the class at the same time. The `.card-focuscontent` instances should also just be changed to use `.cardBox.show-focus` because there's really no need for another class that has the same function.

Reply:
It turned out to be not so simple.

The code controls the focus indication style. For example, in `components/itemidentifier/itemidentifier.js` class `cardBox-focustransform` added in TV mode for "not slow" browser (and at some places not for Edge). On "slow" browsers the color frame is used. The same in
```
components/imageeditor/imageeditor.js
components/cardbuilder/chaptercardbuilder.js
components/cardbuilder/cardBuilder.js
components/imagedownloader/imagedownloader.js
```
Some "simple focus" marker (like `layout-tv`) is needed to save current logic. Maybe it should be configurable like a layout.

**Focus related classes (and possible replacement)**
- `cardBox-focustransform` ~ `.focus-cool .cardBox.show-focus`
Applied to `cardBox` (in some places for "fast" browser only) when transform should be used.

- `card-focusscale`
Applied to `cardBox`. Only appended, no reference in CSS and JS. Safe to remove.

- `card-focuscontent` ~  `.outerBorder .cardBox.show-focus` / `.innerBorder .cardBox.show-focus > .cardScalable`
Set border color of focused card.
In major it is applied to `cardBox`.
In `selectserver.js` and `chaptercardbuilder.js` it is applied to `cardScalable`.
In `cardBuilder.js` it is applied to both (`cardBox` and `cardScalable`) in depend of view style ("Poster" or "Poster card").
So it controls who will have a border: inner (`cardScalable`) or outer (`cardBox`).
This one requires "card border style" marker. Or just leave it in place.
Works in combination with `card-focuscontent-large`.

- `card-focuscontent-large` ~ `.focus-simple .outerBorder  .cardBox.show-focus` / `.focus-simple .innerBorder  .cardBox.show-focus > .cardScalable`
Same as `card-focuscontent`. Set border size when transform is not in use.

- `cardBox-withfocuscontent`
Applied to `cardBox`. Only appended, no reference in CSS and JS. Safe to remove.

- `cardBox-withfocuscontent-large` ~ `.focus-simple .cardBox.show-focus`
Applied to `cardBox` for "slow" browser (and Edge). Set margins when transform is not in use.

Reply:
> It turned out to be not so simple.

That's definitely an understatement :laughing: I remember finding all these classes and slowly backing away, thank you for finally documenting all of them.

What about two classes (show-focus and show-animation) to replace these? The show-focus class will work as mentioned above and the show-animation class will be selectively enabled the same as now. We can then move all transform styles and animations under that class and add it below the layout preference in the user settings.

Reply:
> What about two classes (show-focus and show-animation) to replace these?

It seems it will be enough. I tried them on cardBuilder only (it is most "styled").
`show-focus` - focus should be indicated
`show-animation` - (as addition) focus should be "beautiful"
They can be moved to card (button) itself like in emby-button you mentioned, but this complicates the selector.
Class `.visualCardBox` can be used as marker inner/outer border.

- Somewhere in `components/cardbuilder/cardBuilder.js`
```
if (layoutManager.tv) {

    cardBoxClass += ' show-focus'

    if (enableFocusTransfrom) {
        cardBoxClass += ' show-animation'
    }
}
```
- `components/cardbuilder/card.css` (_replaced classes for reference only_)
```
/*.cardBox-withfocuscontent-large {*/
.cardBox.show-focus:not(.show-animation) {
    margin: .4em;
}
 
/*.card-focuscontent-large {*/
.cardBox.show-focus:not(.show-animation).visualCardBox,
.cardBox.show-focus:not(.show-animation):not(.visualCardBox) .cardScalable {
    border: .5em solid transparent;
}
 
/*.cardBox-focustransform {*/
.cardBox.show-animation {
    will-change: transform;
    transition: transform 200ms ease-out;
}
 
/*.card:focus > .cardBox-focustransform {*/
.card:focus > .cardBox.show-animation {
    transform: scale(1.18, 1.18);
}
```
- `components/themes/dark/theme.css` (_replaced classes for reference only_)
```
/*.card:focus .card-focuscontent,*/
/* if 'show-animation' the border has zero size */
.card:focus .cardBox.visualCardBox,
.card:focus .cardBox:not(.visualCardBox) .cardScalable {
    border-color: #00a4dc
}
```

Also I missed `focus-nofocustransform`. It applied to `card` when transform is not in use.
If `show-focus` and `show-animation` will be applied to a card itself, then it could be removed.
```
/*.card-nofocustransform {*/
.card.show-focus:not(.show-animation) {
    contain: layout style paint;
}
```

Reply:
That sounds excellent. I'm not sure I have a preference on where to apply the classes right now, either way is an improvement.

Reply:
For now, this can be as in `cardBuilder.js`
```
var enableFocusTransform = !browser.slow && !browser.edge;
```
Also I think is better to add these classes to the `card` instead of the `cardBox`. In this case you could tune the `card` style too.

Reply:
go for it! :) should it be done in this PR?